features/resource_status.md

```diff
@@ -0,0 +1,220 @@
+---
+
+## Alpha
+
+### Requirements: 
+
+**Design**
+
+- [x] RFC has been approved describing the intention of the feature as well as the user stories behind the feature. 
+
+**Config**
+
+- [x] Explicit user action is required to enable this feature (e.g. a config field, config resource, or installation action). 
+
+> Link to instructions for enabling
+    https://istio.io/latest/docs/ops/diagnostic-tools/istioctl-analyze/
+
+**Docs**
+
+- [x] Reference docs are published to preliminary.istio.io or the Istio wiki.
+- [x] Basic feature docs are published on preliminary.istio.io describing what the feature does, how to use it, and any caveats. 
+- [x] Release notes entries added as appropriate
+- [x] Upgrade notes entries added as appropriate
+
+**Tests**
+
+- [x] Automated integration tests cover core use cases with the feature enabled. 
+- [x] When disabled, the feature does not affect system stability or performance. 
+
+**API**
+
+- [x] Initial API review.
+
+**Approvals**
+
+- [x] The appropriate work group(s) have reviewed and approved promotion of the feature.
+- [x] The TOC has reviewed and approved promotion of the feature as part of the
+	roadmap for a release.
+
+**Promotion**
+
+[//]: # (Once all other items are completed, features.yaml should be updated to promote the feature)
+
+- [ ] [features.yaml](https://github.com/istio/enhancements/blob/master/features.yaml) updated for this feature
+
+---
+
+## Beta
+
+### Requirements: 
+
+**Design**
+
+- [x] Design doc describing the intention of the feature, how it will be
+	implemented, and any thoughts on how to test the feature has been approved by
+	relevant work group leads
+- [x] Feature coverage and test plans written and approved.
+
+**Docs** 
+
+- [x] Documentation on istio.io includes performance expectations; may have caveats. 
+- [x] Documentation on istio.io includes samples/tutorials. 
+- [x] Documentation on istio.io includes appropriate glossary entries. 
+- [x] All new documentation containing user actions includes istio.io tests.
+- [ ] Release notes have been added. 
+- [ ] Upgrade notes have been added. 
+
+**Tests**
+
+- [x] Integration tests cover feature edge cases
+- [x] End-to-end tests cover samples/tutorials
+- [x] Fixed issues have tests to prevent regressions
+- [x] Stability/stress test suite includes coverage for the feature.
+
+    https://github.com/istio/istio/issues/30179
+
+**Performance**
+
+- [x] Feature coverage and test plans written and approved 
+- [x] Tests exist with the feature enabled that can be integrated with our automated performance testing.
+
+**API**
+
+- [ ] TOC has reviewed the API and determined it to be complete. 
+
+**Tooling**
+
+- [ ] Any necessary tooling to use/debug the feature has been implemented and is complete. 
+
+**Bugs**
+
+- [x] Feature has no known major issues.
+    Resolved issues: https://github.com/istio/istio/issues/30179
```

Comment:
I am not sure these are all solved..? Seems a lot of them were just closed for being stale

Reply:
@howardjohn John, thanks for looking at this, I am hoping that this PR https://github.com/istio/istio.io/pull/11121 can be merged into the docs which contains a link to the tests that I have done. The tests and results are here https://github.com/litong01/multicluster-istio/blob/main/perf/charts/README.md

Reply:
@howardjohn I think we've talked about this list awhile back, the one that needs to be resolved is this https://github.com/istio/istio/issues/30200. I've merged one PR which introduced this new parameter ISTIOD_ANALYSIS_INTERVAL to allow a user to control how often to run analyzers. Unless there is any other issue need to be resolved. If you do have other issues wanting to be resolved, I'm happy to look into.

Reply:
I don't think that solves the problem... you cannot reasonably set a high value (say, 10m), since then the analysis info is too outdated. But you cannot set it low (10s), since then it is way to expensive.

Its not clear why we should have code doing a TON of work looking at the exact same config when it isn't changed. All other parts of Istio are reactive (eg when VS changes, push the new VS). This means it is faster and cheaper.

---

* https://github.com/istio/istio/issues/29709
* https://github.com/istio/istio/issues/30512
* https://github.com/istio/istio/issues/30362
* https://github.com/istio/istio/issues/30292
* https://github.com/istio/istio/issues/30126
* https://github.com/istio/istio/issues/30122
* https://github.com/istio/istio/issues/28015

Just briefly going through the list, the following were stale-closed, so I assume they aren't fixed. Anecdotally, I enabled this flag in my cluster and was immediately presenting with 1000s of error messages in Istiod logs, so it seems at least some must be unfixed.

Reply:
@howardjohn yes, based on the current design, it runs every 10 seconds to analyze all the resources in watched namespaces. that is how it was designed. unless we want to make changes to that, these error messages will be present in the logs unfortunately.